linkerd/app/outbound/src/lib.rs

```diff
@@ -68,9 +68,47 @@ impl Config {
     > + Unpin
            + Clone
            + Send {
+        let connect = svc::connect(self.proxy.connect.keepalive).into_inner();
         // Establishes connections to remote peers (for both TCP
         // forwarding and HTTP proxying).
-        svc::connect(self.proxy.connect.keepalive)
+        self.build_tcp_connect_with(connect, local_identity, metrics)
+    }
+
+    pub fn build_tcp_connect_with<C>(
```

Comment:
If we do need, let's call it build_tcp_endpoint or something... and i think we shuold have a build_tcp_connect that only produces the connect layer...